### PR TITLE
Add interpolation? check for RegexpNode

### LINKED ISSUE #4: RegexpNode needs test for interpolation

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #4

### CODE REVIEW

File: lib/rubocop/ast/node/regexp_node.rb [hunk 1]
```diff
@@ -31,6 +31,11 @@ def regopt
       def content
         children.select(&:str_type?).map(&:str_content).join
       end
+
+      # @return [Bool] if regexp contains interpolation
+      def interpolation?
+        children.select(&:begin_type?).any?
```
Comment (marcandre) [inline lib/rubocop/ast/node/regexp_node.rb:37]: Best use `children.any?(&:begin_type?)`; shorter and faster

**Comment by marcandre** [comment]
👍 Looks like a good addition.

**Comment by tejasbubane** [comment]
Unrelated failure on master fixed in https://github.com/rubocop-hq/rubocop-ast/pull/19

File: spec/rubocop/ast/regexp_node_spec.rb [hunk 2]
```diff
@@ -140,4 +140,24 @@
       it { expect(content).to eq("\n.+\n") }
     end
   end
+
+  describe '#has_interpolation?' do
```
Comment (marcandre) [inline spec/rubocop/ast/regexp_node_spec.rb:144]: It's actually `interpolation?`...

Comment (tejasbubane) [reply spec/rubocop/ast/regexp_node_spec.rb]: I renamed the method but forgot here. Changed now.

**Comment by marcandre** [comment]
Great, thanks! 😸

**Pull request merged**